### PR TITLE
gh-104090: Fix unittest collectedDurations resources leak

### PR DESCRIPTION
**Pull request by bityob**
1. Fixed `TestResult.addDuration` method to add only test repr string and not the test object itself, to avoid resources leak

@iritkatriel
 

* Issue: gh-104090

### CODE REVIEW

**Comment by iritkatriel** [comment]
I think it might be better to split this into two PRs (attached to the same issue), one for the unittest fix and the other for the resource tracker change.

**Comment by bityob** [comment]
> I think it might be better to split this into two PRs (attached to the same issue), one for the unittest fix and the other for the resource tracker change.

Done

Second PR: https://github.com/python/cpython/pull/106807

**Comment by iritkatriel** [comment]
CC unittest maintainers. @gpshead @ezio-melotti

See issue for motivation.

**Comment by sunmy2019** [comment]
Is this new entry necessary?

---

> Should this be backported as a bugfix?

+1 from me

File: Misc/NEWS.d/next/Tests/2023-07-16-02-57-08.gh-issue-104090.cKtK7g.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix unittest collectedDurations resources leak
```
Comment (iritkatriel) [inline Misc/NEWS.d/next/Tests/2023-07-16-02-57-08.gh-issue-104090.cKtK7g.rst]:
```suggestion
Avoid creating a reference to the test object in :meth:`~unittest.TestResult.collectedDurations`.
```

**Review by iritkatriel — APPROVED** [review]

**Comment by giampaolo** [comment]
> @giampaolo addDuration was added in 3.12. Should this be backported as a bugfix?

Hi! Yes, I think it should be backported.

**Comment by gpshead** [comment]
thanks for figuring this out!

**Pull request merged**